src/hypertrace/agent/init/__init__.py

```diff
@@ -324,3 +341,10 @@ def _init_otlp_exporter(self) -> None:
             logger.error('Failed to initialize OTLP exporter: exception=%s, stacktrace=%s',
                          err,
                          traceback.format_exc())
+
+    def is_registered(self, module: str) -> bool:
+        '''Is an instrumentation module already registered?'''
+        try:
```

Comment:
I think it is pretty straighforward to check if the key exists and then return the value instead of throwing an exception. In general I would only expect an exception thrown for things I can't control.

Reply:
It may be straightforward, but python throws a KeyError exception if an invalid key is provided and the keys are based on external input. You can see this behavior below.

So, to avoid this disrupting the flow of logic, a try-except clause that returns False on any exception is used.

$ cat test.py
tester = {
  'test1':'test1',
  'test2':'test2',
  'test3':'test3'
}

print(str(tester['test1']))
print(str(tester['test4']))

$ python test.py
test1
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    print(str(tester['test4']))
KeyError: 'test4'